### ISSUE TITLE
FieldBtn, FieldSig are not variable text

### ISSUE THREAD

**Issue by faceless2** [pdf-association/arlington-pdf-model#29]
There's no ability to change the text in a button or signature. Push buttons no, and radiobutton and checkboxes have to come with On/Off appeareances predefined so no again. Signatures, definitely no.

So I think we can drop DA, Q, DS and RV from `FormBtn` and `FormSig`

**Comment by petervwyatt** [comment]
Great comment that also reflects that corresponding spec clarifications/erratas should be done. Arlington is a blind brute force conversion of the spec to a model so if the spec doesn't say you "shall not" have these keys then the model includes them (i.e. it avoids all inferences and assumptions).

So these 4 keys are the entirety of "Table 228 — Additional entries common to all fields containing variable text" in clause "12.7.4.3 Variable text". AFAICT there are no explicit requirements anywhere in "12.7 Forms" that limit those keys to specific kinds of Fields, and the definition of "variable text" in 12.7.4.3 is hand-wavy at best (e.g. the NOTE gives the example of ECMAscript) and is not sufficiently prescriptive to limit things. 

Can you find any explicit or implicit requirements anywhere below 12.7?

**Comment by faceless2** [comment]
Oh yes, I'm building a mental list of things that could be clarified in the spec from all of this, and variable text is one of them. There's a paragraph description of when it's applied in 12.7.4.3 but it would be helpful to add explicit notes to each field to say "it applies here". For instance, non-combo Choice fields: the "contents and properties of a field are known in advance" is arguably true, but the viewer will still regenerate the appearance stream as the drop-down value changes, so it's variable text.

Another potential spec clarification is inheritance for Fields - they inherit from the Form, despite not having a `Parent` pointer to the Form. This isn't stated in the text on Field inheritance in 12.7.4.1, just that they inherit from "its parent".

We hadn't added this special type of inheritance to our Arlington-model implementation, which meant all our buttons were in error due to no DA. Fixed, but I opened the issue anyway as I think it needs clarifying.

To answer your specific question, I think the best I can do is the second paragraph of 12.7.4.3

> "In such cases, the PDF document cannot provide a statically defined appearance stream for displaying the field. Instead, the PDF processor **shall** construct an appearance stream dynamically at rendering time

Buttons ship with AP streams with statically defined appearance streams, and these won't be regenerated by the viewer as the value changes - the "shall" in that paragraph is not applied. So I'd argue they're not variable text based on this, and the requirements from Table 228 don't apply.
